src/unitxt/metrics.py

```diff
@@ -1430,3 +1431,116 @@ def _compute(
             for k in self.k_list:
                 result[self.score_name(measure_name, k)] = measure_array[min(k, max_k)]
         return result
+
+
+class GroupedInstanceMetric(GlobalMetric):
+    """GlobalMetric calculated as a function on grouped instances (by additional_inputs), then averaged.
+
+    class for GlobalMetrics where each instance belongs to a group defined by the combined values of one or more other variables,
+    specified by the dict input in additional_inputs (e.g. an ID and GROUP variable).
+    each instance score is determined by a function (specified by `metric'), and within each group, the overall score
+    is calculated by applying another function (specified by `group_score_func') across the instance scores.  Then the
+    overall score is calculated by the mean of these group_score_func scores.  For instance, if group_score_func = median and
+    metric = Accuracy, we calculate the Accuracy of each instance, take the median within each group, then the mean of
+    the medians (excluding any NaN values)
+    Note: the implementation could be more efficient, because the instance accuracies only need to be calculated once
+    for the global score and in the confidence interval
+    """
+
+    from statistics import mean
+
+    metric = Accuracy
+    group_score_func = mean
+    group_score_name = "mean"
+    main_score = "mean"
+
+    def compute(
+        self,
+        references: List[List[Any]],
+        predictions: List[Any],
+        additional_inputs: List[Dict],
+    ) -> dict:
+        from collections import defaultdict
+
+        group_to_predictions_and_references = defaultdict(lambda: [[], [], []])
+        for reference, pred, inputs_dict in zip(
+            references, predictions, additional_inputs
+        ):
+            # allow any number of columns to be used as the identifier
+            keyname = tuple(inputs_dict.values())
```

Comment:
I think the user needs to explicitly  specify which are fields from additional_inputs to take.  There could be many which are irrelevant.

Reply:
In cards, is there a way for the user to specify which fields are relevant to be included in additional_inputs?  I assume perhaps there are cases where additional_inputs fields could be used for calculation of the accuracy metric but NOT the grouping (which the aggregation function operates on).  I assumed that all fields in additional_inputs would be used for grouping.  What did you envision?